src/services/transaction_watcher_service.ts

```diff
@@ -0,0 +1,116 @@
+import { SupportedProvider, Web3Wrapper } from '@0x/web3-wrapper';
+import { Connection, Repository } from 'typeorm';
+
+import { TX_WATCHER_POLLING_INTERVAL_IN_MS } from '../constants';
+import { TransactionEntity } from '../entities';
+import { logger } from '../logger';
+import { TransactionStates } from '../types';
+import { utils } from '../utils/utils';
+
+export class TransactionWatcherService {
+    private readonly _transactionRepository: Repository<TransactionEntity>;
+    private readonly _provider: SupportedProvider;
+    private readonly _web3Wrapper: Web3Wrapper;
+
+    constructor(dbConnection: Connection, provider: SupportedProvider) {
+        this._transactionRepository = dbConnection.getRepository(TransactionEntity);
+        this._provider = provider;
+        this._web3Wrapper = new Web3Wrapper(this._provider);
+    }
+    public async startAsync(): Promise<void> {
+        while (true) {
+            // finding transactions that need updating for which we have missed
+            // notifications from blocknative.
+            logger.trace('syncing transaciton status');
+            try {
+                await this.syncTransactionStatusAsync();
+            } catch (err) {
+                logger.error(`order watcher failed to sync transaction status`, { err: err.stack });
+            }
+            await utils.delay(TX_WATCHER_POLLING_INTERVAL_IN_MS);
+        }
+    }
+    public async syncTransactionStatusAsync(): Promise<void> {
+        await this._syncTransactionStatusAsync();
+    }
+
+    public async _syncTransactionStatusAsync(): Promise<void> {
+        const transactionsToCheck = await this._transactionRepository.find({
+            where: [
+                { status: TransactionStates.Unsubmitted },
+                { status: TransactionStates.Submitted },
+                { status: TransactionStates.Mempool },
+                { status: TransactionStates.Stuck },
+            ],
+        });
+        logger.trace(`found ${transactionsToCheck.length} transactions to check status`);
+        for (const tx of transactionsToCheck) {
+            await this._findTransactionStatusAndUpdateAsync(tx);
+        }
+    }
+
+    private async _findTransactionStatusAndUpdateAsync(tx: TransactionEntity): Promise<TransactionEntity> {
+        // TODO(oskar) - we are checking expiry inline, because TypeORM failed
+        // to work with LessThanOrEqual:
+        // const now = new Date();
+        // const transactionsToCheck = await this._transactionRepository.find({
+        //     where: [
+        //         {
+        //             status: TransactionStates.Mempool,
+        //             expiresAt: LessThanOrEqual(now),
+        //         },
+        //         {
+        //             status: TransactionStates.Submitted,
+        //             expiresAt: LessThanOrEqual(now),
```

Comment:
Does LessThan work? We could also use that 🤷

Reply:
No it doesn't work for any of these methods. I also tried to use "format" and specific strings, but that still didn't work properly, I'll leave a reference to a Github issue.